packages/messenger-widget/src/components/AddConversation/bl.ts

```diff
@@ -0,0 +1,193 @@
+    resetInputFieldClass();
+};
+
+// method to add new conversation in contacts
+export const addContact = async (
+    name: string,
+    state: GlobalState,
+    dispatch: React.Dispatch<Actions>,
+    resetName: Function,
+    showErrorMessage: Function,
+    resetInputFieldClass: Function,
+) => {
+    // start loader
+    dispatch({
+        type: ModalStateType.LoaderContent,
+        payload: 'Adding contact...',
+    });
+
+    startLoader();
+
+    // check if its a valid address or a valid name
+    const { normalizedAccountName, check } = await isEnsNameValid(name, state);
+
+    if (!check) {
+        showErrorMessage(true, 'Invalid ENS name');
+        closeLoader();
+        return;
+    }
+
+    // check if contact exists in hidden contact
+    const hiddenContact = state.userDb?.hiddenContacts.find(
+        (contact) =>
+            normalizeEnsName(contact.ensName) === normalizedAccountName,
+    );
+
+    // unhide contact
+    if (hiddenContact && state.accounts.contacts) {
+        // update hidden contact
+        updateHiddenContact(
+            state,
+            dispatch,
+            normalizedAccountName,
+            hiddenContact.aka,
+        );
+
+        // update states
+        updateStates(
+            dispatch,
+            resetName,
+            showErrorMessage,
+            resetInputFieldClass,
+        );
+    } else {
+        // check if contact already exists else create new conversation
+        const doesExists = state.userDb?.conversations.has(
+            normalizedAccountName,
+        );
+
+        if (doesExists) {
+            showErrorMessage(true, 'Contact already exists');
+            closeLoader();
+            return;
+        } else {
+            // update states
+            updateStates(
+                dispatch,
+                resetName,
+                showErrorMessage,
+                resetInputFieldClass,
+            );
+
+            // create empty conversation
+            dispatch({
+                type: UserDbType.createEmptyConversation,
+                payload: normalizedAccountName,
+            });
+        }
+    }
+};
+
+// updates states and closes modal
+const updateStates = (
+    dispatch: React.Dispatch<Actions>,
+    resetName: Function,
+    showErrorMessage: Function,
+    resetInputFieldClass: Function,
+) => {
+    // close the modal
+    closeConversationModal(resetName, showErrorMessage, resetInputFieldClass);
+
+    // set left view to contacts
+    dispatch({
+        type: UiViewStateType.SetSelectedLeftView,
+        payload: LeftViewSelected.Contacts,
+    });
+
+    // set right view to chat
+    dispatch({
+        type: UiViewStateType.SetSelectedRightView,
+        payload: RightViewSelected.Chat,
+    });
+};
+
+// updates hidden contact states
+const updateHiddenContact = (
+    state: GlobalState,
+    dispatch: React.Dispatch<Actions>,
+    normalizedAccountName: string,
+    aka: string | undefined,
+) => {
+    if (!aka) {
```

Comment:
what is aka?

Reply:
It is a string to handle hidden contacts which is used in **DM3 storage library**. This property is used in the UserDB named interface.